packages/calypso-e2e/src/lib/components/sidebar-component.ts

```diff
@@ -26,32 +30,27 @@ export class SidebarComponent {
 	}
 
 	/**
-	 * Waits for the wrapper of the sidebar to be initialized on the page, then returns the element
-	 * handle for that sidebar.
-	 *
-	 * @returns the ElementHandle for the sidebar
+	 * Waits for the WordPress.com Calypso sidebar to be ready on the page.
 	 */
-	async waitForSidebarInitialization(): Promise< ElementHandle > {
+	async waitForSidebarInitialization(): Promise< void > {
 		await this.page.waitForLoadState( 'load' );
-
-		// wait for active promotions to load because they can push the sidebar down, changing items positions
-		await this.page
-			.waitForResponse( ( r ) => Boolean( r.url().match( /active-promotions/i ) ), {
-				timeout: 5000,
-			} )
-			.then( () => {
-				// let active promotions render
-				return this.page.waitForTimeout( 50 );
-			} )
-			.catch( () => console.log( 'Active promotions were not requested' ) );
-
-		const sidebarElementHandle = await this.page.waitForSelector( selectors.sidebar );
-
-		await sidebarElementHandle.waitForElementState( 'stable' );
-
-		return sidebarElementHandle;
+		const sidebarLocator = this.page.locator( selectors.sidebar );
+		await sidebarLocator.waitFor( { state: 'visible' } );
+
+		// If the sidebar is collapsed (via the Collapse Menu toggle),
+		// re-expand the sidebar.
+		if ( await this.sidebarIsCollapsed() ) {
+			const sidebarCollapseToggle = this.page.locator( selectors.linkWithText( 'Collapse menu' ) );
+			await sidebarCollapseToggle.dispatchEvent( 'click' );
```

Comment:
Any reason for pulling this out here instead of in the `else if` below?

Since this is only relevant to desktop viewports, I personally think the logical flow makes a little more since included in the `else if` below. That logic block is then essentially concerned with doing whatever needs to be done to expand the sidebar before we carry on!

Totally optional, but just wanted to throw that perspective out there!

Reply:
I assume the `else if` is referring to Lines 67-69 in the original code? (I'm using side-by-side view).

My reason for putting this here is to simply prep the Sidebar to be in the expected state prior to the navigation steps occurring.

I didn't want to put it behind the `if/else` clause the original code has, because it makes the assumption that mobile and a minified sidebar cannot exist together. While that may be the case now, I don't think we should codify behavior if we don't have to.

Note the above is different from switching behavior based on the `VIEWPORT_SIZE` as the viewport size is binary: "it's either in mobile view, or not" or "it's in tablet view, or not". 

Not sure if I explained this well, but hope it makes sense.

Reply:
Sure thing, that makes a lot of sense! Thanks for explaining! 🙂